inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp

```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
-                    const cv::gapi::fluid::View & d,
-                    cv::gapi::fluid::Buffer & out) {
-        GAPI_DbgAssert(is_cv_type_in_list<merge_supported_types>(out.meta().depth));
+    static inline Unit map(double ratio, int start, int max, int outCoord) {
+        int s = cvFloor(outCoord*ratio);
+        float f = static_cast<float>((outCoord+1) - (s+1)/ratio);
+        f = f <= 0 ? 0.f : f - cvFloor(f);
 
-        const auto rowFunc = type_dispatch<merge_supported_types>(out.meta().depth, cv_type_id{}, typed_merge_row<isa_tag_t, 4>{}, nullptr);
-        for (int l = 0; l < out.lpi(); l++) {
-            rowFunc({ a.InLineB(l), b.InLineB(l), c.InLineB(l), d.InLineB(l) }, out.OutLineB(l), a.length());
-        }
+        Unit u;
+
+        u.index0 = std::max(s - start, 0);
+        u.index1 = ((f == 0.0) || s + 1 >= max) ? s - start : s - start + 1;
+
+        u.alpha0 = 1.0f - f;
+        u.alpha1 =        f;
+
+        return u;
     }
 };
+}  // namespace areaUpscale32f
 
-template<typename KT, typename TL>
-static inline void callRowFunc(uint8_t* dst[], const uint8_t* src0[],
-                               const uint8_t* src1[], const short alpha[],
-                               const short clone[], const short mapsx[],
-                               const short beta[], uint8_t tmp[], const Size& inSz,
-                               const Size& outSz, const int lpi, const int length, const int depth) {
-    const auto rowFunc = type_dispatch<TL>(depth, cv_type_id{}, KT{}, nullptr);
+template<typename Mapper>
+static void initScratchArea(const cv::GMatDesc& in, const Size& outSz,
+                            cv::gapi::fluid::Buffer &scratch) {
+    using Unit = typename Mapper::Unit;
+    using alpha_type = typename Mapper::alpha_type;
+    using index_type = typename Mapper::index_type;
 
-    GAPI_DbgAssert(rowFunc);
+    // compute the chunk of input pixels for each output pixel,
+    // along with the coefficients for taking the weigthed sum
 
-    rowFunc(dst, src0, src1, alpha, clone, mapsx, beta, tmp, inSz, outSz, lpi, length);
-}
+    Size inSz = in.size;
+    Mapper mapper(inSz.width, outSz.width);
 
-template<typename KT, typename TL>
-static inline void callRowFunc(float* dst[], const float* src0[], const float* src1[],
-                               const float alpha[], const float clone[], const int mapsx[],
-                               const float beta[], float tmp[], const Size& inSz,
-                               const Size& outSz, const int lpi, const int length, const int depth) {
-    const auto rowFunc = type_dispatch<TL>(depth, cv_type_id{}, KT{}, nullptr);
+    std::vector<Unit> xmaps(outSz.width);
+    int  maxdif = 0;
 
-    GAPI_DbgAssert(rowFunc);
+    for (int w = 0; w < outSz.width; w++) {
+        Unit map = mapper.map(w);
+        xmaps[w] = map;
 
-    rowFunc(dst, src0, src1, alpha, mapsx, beta, inSz, outSz, lpi, length);
-}
+        int dif = map.index1 - map.index0;
```

Comment:
the more shorter version can be:

     maxdif = std::max(maxdif, map.index1 - map.index0);

Reply:
Done.